src/app.js

```diff
@@ -274,36 +279,70 @@ fluid.onUncaughtException.addListener(function (err) {
 /**
  * Component that contains an Electron Tray.
  */
-
 fluid.defaults("gpii.app.tray", {
-    gradeNames: "fluid.component",
+    gradeNames: "fluid.modelComponent",
     members: {
         tray: {
             expander: {
                 funcName: "gpii.app.makeTray",
-                args: ["{that}.options.icon"]
+                args: ["{that}.model.icon"]
             }
         }
     },
+    icons: {
+        keyedIn: "icons/gpii-color.ico",
+        keyedOut: "icons/gpii.ico"
+    },
     components: {
         menu: {
             type: "gpii.app.menuInApp"
         }
     },
+    model: {
+        keyedInUserToken: null,
+        icon: "{that}.options.icons.keyedOut"
+    },
+    modelRelay: {
+        keyedInUserToken: {
+            target: "icon",
+            singleTransform: {
```

Comment:
`fluid.transforms.condition` seems not to working properly with falsy values - it returns `undefined` if the condition is `null` (which is the case in the keyed out state). When the user is keyed in, the `true` icon is used; but after keyed out, the icon is left unchanged... An inner `free transform` for the condition should do the job, but it's not improvement on the initial version. @michelled, what do you do in such cases?

Reply:
I recommend using the valueMapper transform which can map arbitrary input values including null and undefined. Sorry I can't link to our docs site which is inaccessible today but here is the relevant section of the source .md : https://github.com/fluid-project/infusion-docs/blob/master/src/documents/ModelTransformationAPI.md#mapping-based-on-input-value-fluidtransformsvaluemapper

@kaspermarkus, do you recall why we made this special treatment of null in the condition transform?

Reply:
Here's that link again - http://docs.fluidproject.org/infusion/development/ModelTransformationAPI.html#mapping-based-on-input-value-fluid-transforms-valuemapper-